packages/compiler-cli/src/ngtsc/transform/src/api.ts

```diff
@@ -37,6 +37,22 @@ export enum HandlerPrecedence {
   WEAK,
 }
 
+/**
+ * A set of options which can be passed to a `DecoratorHandler` by a consumer, to tailor the output
+ * of compilation beyond the decorators themselves.
+ */
+export enum HandlerFlags {
+  /**
+   * No flags set.
+   */
+  NONE = 0x0,
+
+  /**
+   * Indicates that this decorator should fully inherited from its parent at runtime.
```

Comment:
```suggestion
   * Indicates that this decorator should be fully inherited from its parent at runtime.
```

Also, I am confused by this sentence. As I understand it this decorator is almost empty, so the _decorator_ itself should not be inherited, rather the compiled definition should be pieced together at runtime by inheriting all info from the parent def.

I am not sold on the name, but also can't come up with anything better.

Having some more details on why this flags exists would also be super helpful for future reference.